idea/src/org/jetbrains/kotlin/idea/inspections/LiftReturnOrAssignmentInspection.kt

```diff
@@ -24,80 +24,94 @@ import org.jetbrains.kotlin.idea.intentions.branchedTransformations.isElseIf
+                if (expression.lineCount() > LINES_LIMIT) return
+                if (expression.isElseIf()) return
 
-                    val foldableReturns = BranchedFoldingUtils.getFoldableReturns(expression)
-                    if (foldableReturns?.isNotEmpty() == true) {
-                        val hasOtherReturns = expression.anyDescendantOfType<KtReturnExpression> { it !in foldableReturns }
-                        val isSerious = !hasOtherReturns && foldableReturns.size > 1
-                        val verb = if (isSerious) "should" else "can"
+                val foldableReturns = BranchedFoldingUtils.getFoldableReturns(expression)
+                if (foldableReturns?.isNotEmpty() == true) {
+                    val hasOtherReturns = expression.anyDescendantOfType<KtReturnExpression> { it !in foldableReturns }
+                    val isSerious = !hasOtherReturns && foldableReturns.size > 1
+                    val verb = if (isSerious) "should" else "can"
+                    val description = "Return $verb be lifted out of '${keyword.text}'"
+                    holder.registerProblemWithoutOfflineInformation(
+                        expression,
+                        description,
+                        isOnTheFly,
+                        if (isSerious) ProblemHighlightType.GENERIC_ERROR_OR_WARNING
+                        else ProblemHighlightType.INFORMATION,
+                        keyword.textRange?.shiftRight(-expression.startOffset),
+                        LiftReturnOutFix(keyword.text)
+                    )
+                    foldableReturns.forEach {
                         holder.registerProblemWithoutOfflineInformation(
-                                keyword,
-                                "Return $verb be lifted out of '${keyword.text}'",
-                                isOnTheFly,
-                                if (isSerious) ProblemHighlightType.GENERIC_ERROR_OR_WARNING
-                                else ProblemHighlightType.INFORMATION,
-                                LiftReturnOutFix(keyword.text)
-                        )
-                        return
-                    }
-                    val assignmentNumber = BranchedFoldingUtils.getFoldableAssignmentNumber(expression)
-                    if (assignmentNumber > 0) {
-                        val verb = if (assignmentNumber > 1) "should" else "can"
-                        holder.registerProblemWithoutOfflineInformation(
-                                keyword,
-                                "Assignment $verb be lifted out of '${keyword.text}'",
-                                isOnTheFly,
-                                if (assignmentNumber > 1) ProblemHighlightType.GENERIC_ERROR_OR_WARNING
-                                else ProblemHighlightType.INFORMATION,
-                                LiftAssignmentOutFix(keyword.text)
+                            expression,
```

Comment:
May be we should extract this fragment to some local function, to avoid code duplication? What do you think, @t-kameyama ?